systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/OauthKafkaClient.java

```diff
@@ -4,29 +4,257 @@
 import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.IntPredicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-public class OauthKafkaClient implements IKafkaClient {
+import static io.strimzi.systemtest.kafkaclients.externalClients.KafkaClientProperties.fixBadlyImportedAuthzSettings;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+public class OauthKafkaClient implements IKafkaClient<Future<Integer>> {
+
+    private Vertx vertx = Vertx.vertx();
+    private static final Logger LOGGER = LogManager.getLogger(KafkaClient.class);
+
+    private String caCertName;
+    private String clientId;
+    private String clientSecretName;
+    private String oauthTokenEndpointUri;
+
+    public Future<Integer> sendMessages(String topicName, String namespace, String clusterName, int messageCount) throws RuntimeException, IOException {
+        return sendMessages(topicName, namespace, clusterName, messageCount, Constants.GLOBAL_CLIENTS_TIMEOUT);
+    }
 
     @Override
-    public Future<Integer> sendMessages(String topicName, String namespace, String clusterName, int messageCount) {
-        throw new UnsupportedOperationException();
+    public Future<Integer> sendMessages(String topicName, String namespace, String clusterName, int messageCount,
+                                        long timeoutMs) throws RuntimeException, IOException {
+        String clientName = "sender-plain-" + clusterName;
+        vertx = Vertx.vertx();
+        CompletableFuture<Integer> resultPromise = new CompletableFuture<>();
+
+        IntPredicate msgCntPredicate = x -> x == messageCount;
+
+        vertx.deployVerticle(new Producer(KafkaClientProperties.createOauthProducerProperties(namespace, clusterName,
+                this.clientId, this.clientSecretName, this.oauthTokenEndpointUri), resultPromise, msgCntPredicate,
+                topicName, clientName));
+
+        try {
+            resultPromise.get(timeoutMs, TimeUnit.MILLISECONDS);
+        } catch (Exception e) {
+            resultPromise.completeExceptionally(e);
+        }
+        vertx.close();
+        return resultPromise;
+    }
+
+    public Future<Integer> sendMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername,
+                                           int messageCount, String securityProtocol) throws RuntimeException, IOException {
+        return sendMessagesTls(topicName, namespace, clusterName, kafkaUsername, messageCount, securityProtocol,
+                Constants.GLOBAL_CLIENTS_TIMEOUT);
     }
 
     @Override
-    public Future<Integer> sendMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername, int messageCount, String securityProtocol) {
-        throw new UnsupportedOperationException();
+    public Future<Integer> sendMessagesTls(String topicName, String namespace, String clusterName, String kafkaUsername,
```

Comment:
This difference between this method and `sendMessages` above appear to be minimal: Can you factor out a common method?